### PR TITLE
Add a real matrix to Static analysis workflow

### PR DESCRIPTION
**Pull request by szepeviktor**
- Added a real 3×2×1×1 matrix + 3 "includes"
- Move database options to `db-options`
- PHP 8.0 + MariaDB was excluded - Should I exclude it now?

### CODE REVIEW

**Comment by szepeviktor** [comment]
@staabm You may copy it to the unit tests workflow.

**Comment by staabm** [comment]
> PHP 8.0 + MariaDB was excluded - Should I exclude it now?

If green - we can keep them

---


Could you have a look at the failling jobs?

File: .github/workflows/phpstan.yml [hunk 1]
```diff
@@ -60,13 +53,13 @@ jobs:
...
     services:
-      mysql:
+      database:
         image: ${{ matrix.db-image }}
         env:
           MYSQL_ROOT_PASSWORD: root
         ports:
           - 3306:3306
-        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3
+        options: ${{ startsWith(matrix.db-image, 'mariadb') && '--health-cmd="healthcheck.sh --connect --innodb_initialized" --health-interval=1s --health-timeout=10s --health-retries=60' || '--health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=5s --health-retries=3' }}
```
Comment (staabm) [inline .github/workflows/phpstan.yml:62]: can we move the command into the matrix, where we define the db image?

**Comment by szepeviktor** [comment]
> Could you have a look at the failling jobs?

Those job are in the unit test workflow. I leave them 101% up to you.

File: .github/workflows/phpstan.yml (hunk 1 above)
Comment (szepeviktor) [reply .github/workflows/phpstan.yml:62]: Yes ❗

Comment (szepeviktor) [reply .github/workflows/phpstan.yml]: Done 🍏

image

File: .github/workflows/phpstan.yml [hunk 2]
```diff
@@ -19,6 +19,7 @@ jobs:
           - "8.0"
           - "8.1"
           - "8.2"
+          - "8.3"
```
Comment (szepeviktor) [inline .github/workflows/phpstan.yml:22]: This is too daring 👴🏻 for viktor.

Comment (staabm) [reply .github/workflows/phpstan.yml:22]: I agree. I just want to see how the matrix explodes with this additional version, which we need in the near future. will revert afterwards

Comment (szepeviktor) [reply .github/workflows/phpstan.yml:22]:
```
tomasvotruba/unused-public 0.2.0.72 requires php ^7.2 || 8.0.*
 -> your php version (8.3.0-dev) does not satisfy that requirement.
```

Comment (staabm) [reply .github/workflows/phpstan.yml:22]: I guess this will fix it: https://github.com/staabm/phpstan-dba/pull/619

**Comment by szepeviktor** [comment]
@TomasVotruba uses this mysterious `.72` tail on his version.

```
tomasvotruba/unused-public 0.2.0.72 requires php ^7.2 || 8.0.*
 -> your php version (8.3.0-dev) does not satisfy that requirement.
```

How to get the `.72`-less version?

**Comment by staabm** [comment]
I think the problem is that there currently is no nette/utils release which supports PHP8.3

https://github.com/nette/utils/pull/297

**Comment by staabm** [comment]
thank you

**Pull request merged**